security/pkg/platform/client.go

```diff
@@ -45,7 +47,10 @@ func NewClient(platform, rootCertFile, keyFile, certChainFile string) (Client, e
 	case "gcp":
 		// Temporarily disable ID token authentication on CSR API.
 		// [TODO](myidpt): enable when the Citadel authz can work correctly.
-		return nil, fmt.Errorf("GCP credential authentication in CSR API is disabled")
+		//return nil, fmt.Errorf("GCP credential authentication in CSR API is disabled")
+		// TODO(pitlv2109): Make `Aud` configurable.
+		return NewGcpClientImpl(rootCertFile, "istioca.googleapis.com:443",
```

Comment:
This prevents one use case (where this "gcp" is originally used for): Citadel as CA, VM workload on GCE, Citadel verifies instance doc jwt and issue certificate. We might still want to support that use case.

Let's leave room for that.

Reply:
This is intentional. It only supports Google CA for now. We can discuss how it works with Citadel as a CA.

Reply:
fyi @myidpt 

Well we used to have GCE + Citadel for a while, we disabled it because we didn't come up with a authorization check.

The platform param used to indicate where the VM workloads are running, but now we conflate that as well as where the CA backend is running. The right way should be using different param to indicate the CA backend. Otherwise, it can be confusing in future when we introduce other combinations.

Or we should totally change the meaning of platform at all. Update the document to points out it's used to configure CA backend now, and disable AWS case.

@myidpt thoughts?